src/services/API/RootClient.js

```diff
@@ -52,9 +52,13 @@ client.interceptors.request.use(
 // handle errors in responses for API requests
 client.interceptors.response.use(
   (config) => {
+    // reset reAuthenticationState to it's original value whenever any request returns
+    // a non error response code. Make sure not to do this for refresh token call
+    if (config.config.url != refreshTokenEndpoint)
```

Comment:
should additionally place a check that the reAuthenticationState is not started

Reply:
```suggestion
    if (config.config.url != refreshTokenEndpoint && store.state.auth.reAuthenticationState != "not-started")
```